clients/src/main/java/org/apache/kafka/common/metrics/IntGaugeSuite.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.metrics;
+
+import org.apache.kafka.common.MetricName;
+import org.slf4j.Logger;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Manages a suite of integer Gauges.
+ */
+public class IntGaugeSuite<K> implements AutoCloseable {
+    private final Logger log;
+    private final String suiteName;
+    private final Metrics metrics;
+    private final Object lock;
+    private final Function<K, MetricName> metricNameCalculator;
+    private final int maxEntries;
+    private final Map<K, StoredIntGauge> map;
+    private boolean closed;
+
+    private class StoredIntGauge implements Gauge<Integer> {
```

Comment:
It turns out to be relatively complicated to generalize this to any numeric type.  Java has a parent interface `java.lang.Number`, but it doesn't provide much functionality at all.  We would need at minimum subtraction, addition, and checking for zero.  Maybe eventually we will want to support Long here, but I think we can take the complexity hit then rather than doing it now.